### PR TITLE
Add support for custom actions

### PR DESCRIPTION
**Pull request by cwells**
#### Pull Request (PR) description

Allow creation of custom actions.

### CODE REVIEW

File: manifests/action.pp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# == Define: fail2ban::action
+#
+define fail2ban::action (
+  String $actionban = undef,
+  String $actionunban = undef,
+
+  Optional[String] $config_file_owner = $fail2ban::config_file_owner,
```
Comment (bastelfreak) [inline manifests/action.pp]: can you please align all the `=` chars and enforce a minimal string length of 1?
```suggestion
  Optional[String[1]] $config_file_owner = $fail2ban::config_file_owner,
```

File: manifests/action.pp [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  Optional[String] $config_file_mode = $fail2ban::config_file_mode,
+  Optional[String] $config_file_source = $fail2ban::config_file_source,
+  Optional[String] $config_file_notify = $fail2ban::config_file_notify,
+  Optional[String] $config_file_require = $fail2ban::config_file_require,
+) {
+
+  file { "custom_action_${name}":
+    ensure  => file,
+    path    => "${::fail2ban::params::config_dir_path}/action.d/${name}.conf",
+    content => epp('fail2ban/common/custom_action.conf.epp',
```
Comment (bastelfreak) [inline manifests/action.pp]: `epp()` \o/

File: manifests/action.pp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  Optional[String] $config_file_group = $fail2ban::config_file_group,
+  Optional[String] $config_file_mode = $fail2ban::config_file_mode,
+  Optional[String] $config_file_source = $fail2ban::config_file_source,
+  Optional[String] $config_file_notify = $fail2ban::config_file_notify,
+  Optional[String] $config_file_require = $fail2ban::config_file_require,
+) {
+
+  file { "custom_action_${name}":
+    ensure  => file,
+    path    => "${::fail2ban::params::config_dir_path}/action.d/${name}.conf",
```
Comment (bastelfreak) [inline manifests/action.pp]: please don't add leading `::`, they aren't required anymore since puppet 4.

**Comment by bastelfreak** [comment]
Hi @cwells, thanks for the PR! Can you take a look at the inline comments I made?

File: manifests/action.pp (hunk 2 above)
Comment (cwells) [reply manifests/action.pp]: it's not missing a paren.

File: manifests/action.pp (hunk 3 above)
Comment (cwells) [reply manifests/action.pp]: Just fyi, most of the things you are complaining about were copied/pasted from jail.pp.

File: manifests/action.pp (hunk 2 above)
Comment (alexjfisher) [reply manifests/action.pp]: @cwells He was just happy to see `epp` instead of `template` (using erb)

File: manifests/config.pp [hunk 4]
```diff
@@ -34,6 +35,11 @@
     create_resources('fail2ban::jail', $config_custom_jails)
   }
 
+  # Custom actions definition
+  if $config_custom_actions {
+    create_resources('fail2ban::action', $config_custom_actions)
```
Comment (alexjfisher) [inline manifests/config.pp]: We should really move away from using `create_resources`.

**Comment by sternj** [comment]
@cwells why was this closed? I'm not seeing a merge commit to any branch

**Comment by cwells** [comment]
It's more productive for me to just maintain a fork of this project.

**Pull request closed without merging**